src/components/MultiHopTrade/hooks/useGetTradeQuotes/useGetTradeQuotes.tsx

```diff
@@ -222,24 +231,57 @@ export const useGetTradeQuotes = () => {
       const potentialAffiliateBps = feeBpsBeforeDiscount.toFixed(0)
       const affiliateBps = feeBps.toFixed(0)
 
-      const updatedTradeQuoteInput: GetTradeQuoteInput | undefined = await getTradeQuoteInput({
-        sellAsset,
-        sellAccountNumber,
-        receiveAccountNumber,
-        sellAccountType: sellAccountMetadata.accountType,
-        buyAsset,
-        wallet,
-        receiveAddress,
-        sellAmountBeforeFeesCryptoPrecision: sellAmountCryptoPrecision,
-        allowMultiHop: true,
-        affiliateBps,
-        potentialAffiliateBps,
-        // Pass in the user's slippage preference if it's set, else let the swapper use its default
-        slippageTolerancePercentageDecimal: userSlippageTolerancePercentageDecimal,
-        pubKey: isLedger(wallet) ? fromAccountId(sellAccountId).account : undefined,
-      })
-
-      setTradeQuoteInput(updatedTradeQuoteInput)
+      if (hasWallet) {
```

Comment:
feels like we don't need this if condition either. just pass the args to the swapper where all validation and discrimination is handled.